google-beta/node_config.go

```diff
@@ -238,6 +259,96 @@ func schemaNodeConfig() *schema.Schema {
 					Optional: true,
 					ForceNew: true,
 				},
+
+				"kubelet_config": {
+					Type:     schema.TypeList,
+					Optional: true,
+					MaxItems: 1,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"cpu_manager_policy": {
+								Type:         schema.TypeString,
+								Optional:     true,
+								ValidateFunc: validation.StringInSlice([]string{"static", "default"}, false),
+							},
+							"cpu_cfs_quota": {
+								Type:     schema.TypeBool,
+								Optional: true,
+							},
+							"cpu_cfs_quota_period": {
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+						},
+					},
+				},
+
+				"linux_node_config": {
+					Type:     schema.TypeList,
```

Comment:
While adding extra validation to the schema is generally positive, could it make more sense here to use a `TypeMap` instead? That way users write the keys themselves and we don't need to track name differences with `sysAttrsMap`. We'll also have the same string->string type as the client library, avoiding the need to cast values and letting us simplify the expanders/flatteners considerably.

The downside is that we lose some of the `ValidateFunc` behaviour you've written. When using a map I don't think we can add per-key validation, so examples like `net_core_somaxconn` won't have lower bound checking anymore.

Reply:
Yep, happy to change this for a TypeMap. Will try iterate on this a bit later tonight. Thanks for the feedback so far!

Reply:
fixed